### DISCUSSION TITLE
模块包里面typings.d.ts  文件是做什么用的呢？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/umi#11105] [Q&A]
我用umi创建好一个项目之后，
发现包里面有一个 typings.d.ts 文件：
![image](https://user-images.githubusercontent.com/12727582/236653427-68a164fa-ee69-4d7d-941d-be9b8195ff54.png)

内容如下：
```
declare namespace API {
  interface PageInfo {...}
  interface PageInfo_UserInfo_ {}
  interface Result_PageInfo_UserInfo__ {}
  type UserGenderEnum = 'MALE' | 'FEMALE';
  interface UserInfo {}
  ...
}
```

搜索里面定义的interface，也没有找到其他的地方使用到：
![image](https://user-images.githubusercontent.com/12727582/236653491-d8a93412-368b-40c3-8a80-26e7eb778c75.png)

**Comment by fz6m** [comment]
[`openAPI`](https://github.com/ant-design/ant-design-pro/blob/fd4a4a0183ead9863bcf3c700ae7eb7218660ff2/config/config.ts#L137) 自动生成的。
